src/grader/mutation_test.mli

```diff
@@ -0,0 +1,110 @@
+(* This file is part of Learn-OCaml.
+ *
+ * Copyright (C) 2019 OCaml Software Foundation.
+ * Copyright (C) 2016-2018 OCamlPro.
+ *
+ * Learn-OCaml is distributed under the terms of the MIT license. See the
+ * included LICENSE file for details. *)
+
+type 'a test_result =
+  | Pass
+  | Fail of 'a
+  | Err of exn
+
+(** A mutant is a pair of a string describing the mutant and the
+    mutant function itself.
+*)
+type 'a mutant = string * 'a
+
+(** Run a test (a pair of input and expected output) on a function.
+    The [compare] parameter specifies a comparison function for
+    comparing the expected and actual outputs, and defaults to
+    structural equality ([(=)]).
+*)
+val run_test_against:
+  ?compare: ('b -> 'b -> bool) ->
+  ('a -> 'b) -> ('a * 'b) -> 'b test_result
+
+(** Run a test (a pair of input and expected output) on a mutant.
```

Comment:
Given this first sentence, it is a bit weird not to see a `mutant` as an argument of this function.